view/frontend/web/js/components/steps/TheShippingStep.vue

```diff
@@ -281,6 +295,16 @@ export default {
           console.log('Error with process your Shipping step and process your order - please try again later')
         })
     }
+  },
+  watch: {
+    countryWatcher (newCountry) {
+      this.address.country_id = newCountry.value
+      this.regionWatcher = ''
+      this.onCountryChange()
+    },
+    regionWatcher (newRegion) {
+      this.address.region_id = newRegion.value
+    }
```

Comment:
I think you don't need to use watchers here. IMO regular component data is enough.

Reply:
same as in payment step